src/integrations/watchers/option-titles-watcher.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Watcher for the titles option.
+ *
+ * @package Yoast\YoastSEO\Watchers
+ */
+
+namespace Yoast\WP\SEO\Integrations\Watchers;
+
+use Yoast\WP\SEO\Conditionals\Migrations_Conditional;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+use Yoast\WP\SEO\ORM\Yoast_Model;
+use Yoast\WP\SEO\WordPress\Wrapper;
+
+/**
+ * Represents the option titles watcher.
+ */
+class Option_Titles_Watcher implements Integration_Interface {
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * This is the place to register hooks and filters.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		add_action( 'update_option_wpseo_titles', [ $this, 'check_option' ], 10, 2 );
+	}
+
+	/**
+	 * Returns the conditionals based in which this loadable should be active.
+	 *
+	 * @return array
+	 */
+	public static function get_conditionals() {
+		return [ Migrations_Conditional::class ];
+	}
+
+	/**
+	 * Checks if one of the relevant options has been changed.
+	 *
+	 * @param array $old_value The old value of the option.
+	 * @param array $new_value The new value of the option.
+	 *
+	 * @return bool Whether or not the option has been saved.
+	 */
+	public function check_option( $old_value, $new_value ) {
+		// If this is the first time saving the option, thus when value is false.
+		if ( $old_value === false ) {
+			$old_value = [];
+		}
+
+		if ( ! is_array( $old_value ) || ! is_array( $new_value ) ) {
+			return false;
+		}
+
+		$relevant_keys = $this->get_relevant_keys();
+		$post_types    = [];
+
+		foreach ( $relevant_keys as $post_type => $relevant_option ) {
+			// If both values aren't set they haven't changed.
+			if ( ! isset( $old_value[ $relevant_option ] ) && ! isset( $new_value[ $relevant_option ] ) ) {
+				continue;
+			}
+
+			if ( $old_value[ $relevant_option ] !== $new_value[ $relevant_option ] ) {
+				$post_types[] = $post_type;
+			}
+		}
+
+		if ( empty( $post_types ) ) {
+			return true; // There is nothing to do.
+		}
+
+		$wpdb = Wrapper::get_wpdb();
+
+		$post_types_placeholders = implode( ', ', array_fill( 0, count( $post_types ), '%s' ) );
+		$hierarchy_table         = Yoast_Model::get_table_name( 'Indexable_Hierarchy' );
+		$indexable_table         = Yoast_Model::get_table_name( 'Indexable' );
+
+		$wpdb->query(
```

Comment:
Thinking about how I could write this in Idiorm... the subquery is the problem here?

Reply:
Yes. Tried too much things with no success :(